pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -241,9 +214,18 @@ func (c *clusterRefreshController) refreshClusterUpstreamSpec(cluster *mgmtv3.Cl
 		specMap[key] = value
 	}
 
-	if !updateClusterConfig {
+	if updateClusterConfig {
+		logrus.Debugf("change detected for cluster [%s], updating spec", cluster.Name)
+		// for other cloud drivers, please edit HERE
+		switch cloudDriver {
+		case apimgmtv3.ClusterDriverEKS:
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(specMap, cluster.Spec.EKSConfig)
+			if err != nil {
+				return cluster, err
+			}
```

Comment:
This needs a `ClusterDriverGKE` case added

Reply:
Yes, this is part of the rebase.